### PR TITLE
removing the from nailgun import entities and moving target_sat, module_target_sat

### PR DESCRIPTION
**Pull request by omkarkhatavkar**
#### Key Changes:

- **Direct Imports Removed**: We've cleaned up our codebase by eliminating direct imports from the nailgun package.

- **Using target_sat, module_target_sat, class_target_sat**: Modified the API calls and using the `target_sat.api.` pattern for all API calls.

### CODE REVIEW

**Comment by omkarkhatavkar** [comment]
trigger: test-robottelo

**Comment by omkarkhatavkar** [comment]
trigger: test-robottelo

**Comment by omkarkhatavkar** [comment]
@shweta83 @vsedmik @pondrejk @Griffin-Sullivan Can you please also take a look from the components side as it is touching multiple component tests.

**Comment by Gauravtalreja1** [comment]
@omkarkhatavkar Great job, thanks for this refactor ❤️
Could you add changes for updating fixtures, and setup part done by nailgun entities in CLI/UI/destructive/etc modules too? or are you intending to open a separate PR for each?

**Comment by shweta83** [comment]
@omkarkhatavkar Changes impacting Rocket team components are good. Ack for the rocket team tests except for  @Gauravtalreja1 comment.

**Comment by omkarkhatavkar** [comment]
> @omkarkhatavkar Great job, thanks for this refactor ❤️ Could you add changes for updating fixtures, and setup part done by nailgun entities in CLI/UI/destructive/etc modules too? or are you intending to open a separate PR for each?

@Gauravtalreja1 It will be the seperate PR's as It will be easy to test it out and also review will be smoother

**Review by lpramuk — APPROVED** [review]

**Review by Griffin-Sullivan — APPROVED** [review]
As far as I can tell, none of these components are owned by Platform. Everything looks good to me!

**Review by ColeHiggins2 — APPROVED** [review]
Looks good to me! Nice work

**Comment by JacobCallahan** [comment]
@omkarkhatavkar please rebase and we'll get this in!

File: tests/foreman/api/test_organization.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ def valid_org_data_list():
 
     Note: The maximum allowed length of org name is 242 only. This is an
     intended behavior (Also note that 255 is the standard across other
-    entities.)
+    target_sat.api.)
```
Comment (JacobCallahan) [inline tests/foreman/api/test_organization.py:47]: we can revert this change ;)

**Review by JacobCallahan — CHANGES_REQUESTED** [review]
one minor change

**Comment by omkarkhatavkar** [comment]
trigger: test-robottelo

**Comment by omkarkhatavkar** [comment]
trigger: test-robottelo

**Review by vsedmik — APPROVED** [review]
Looks good, thank you for the PR!

**Comment by omkarkhatavkar** [comment]
trigger: test-robottelo
pod_resources_size: large

**Review by vijaysawant — CHANGES_REQUESTED** [review]

File: tests/foreman/api/test_discoveryrule.py [hunk 2]
```diff
@@ -16,16 +16,35 @@
...
 """
-from fauxfactory import gen_choice, gen_integer
+from fauxfactory import gen_choice, gen_integer, gen_string
 import pytest
 from requests.exceptions import HTTPError
 
 from robottelo.utils.datafactory import valid_data_list
 
 
+@pytest.fixture(scope="module")
```
Comment (vijaysawant) [inline tests/foreman/api/test_discoveryrule.py:26]: To maintain consistancy through out the files, same applicable for line 33, 39
```suggestion
@pytest.fixture(scope='module')
```

**Comment by omkarkhatavkar** [comment]
Hi @JacobCallahan and @vijaysawant, please review. Once confirmed, we can merge without frequent rebases. Thanks!

**Review by JacobCallahan — APPROVED** [review]

**Pull request merged**